Migrate/comments-modal component

b0ink opened the pull request:
Student ID: s222313935

# Description

Migrates the `comments-modal` component from CoffeeScript to TypeScript. This component is used to display images and PDFs from the task comments as a fullscreen modal popup, allowing users to view the content without needing to download or open it in another tab.

## Type of change

- [x] Migration

# How Has This Been Tested?
- Log in as `student_1` and select any unit and any task.
- Send an image and a PDF into the task comments either by manually attaching the files using the bottom left icon or dragging and dropping it over.
- Once uploaded, confirm the `comments-modal` is working correctly by clicking on the attachments in the comments and view images in fullscreen or PDFs embedded into the modal.

## Before (images):
![comments-modal-image-original](https://github.com/user-attachments/assets/c20fbf99-4637-40ba-96e6-519958f7fcbc)

## After (images):
![comments-modal-image-migrated](https://github.com/user-attachments/assets/48e7c756-38df-4c3e-a6a2-925347816788)

## Before (PDFs):
![comments-modal-pdf-original](https://github.com/user-attachments/assets/739a5768-c679-45a4-8a7a-8b5e284d70b5)

## After (PDFs):
![comments-modal-pdf-migrated](https://github.com/user-attachments/assets/85618591-6b7a-4dcd-a862-5b873fb00e33)

## Testing Checklist:

- [x] Tested in latest Chrome
- [x] Tested in latest Safari
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

Component review PR: https://github.com/thoth-tech/documentation/pull/519

s223749059 committed:
refactor: migrate the confirmation-model component

s223749059 committed:
refactor: migrate the confirmation-model component - updates

b0ink committed:
refactor: create initial files for migration of comments-modal

b0ink committed:
refactor: create service file for migration of comments-modal

b0ink committed:
refactor: migrate comments-modal

b0ink committed:
refactor: remove old comments-modal files

b0ink committed:
fix: match comments-modals's original max width

AusBruce commented:
hi, there!

b0ink commented:
I have requested for @AusBruce to review this PR

b0ink committed:
refactor: enforce type safety for data passed to comments-modal component

atharv02-git commented:
# Ontrack Component-Comments-Modal Migration review #519 

## Review by Team Member Atharv

## Component Name: Comments Modal

## Component Review:
Hi Eliya, firstly you've done a great job migrating the "comments-modal" component, so Kudos to that!!!
The code seems to be cleaner, and more maintainable. Honestly, I haven't cloned your repository, because of some issues with my local container setup but looking at the screenshot, the code seems to work absolutely fine.

So, if you could just click on [this link](https://github.com/thoth-tech/documentation/pull/519#issuecomment-2254351056) you could see few suggestions I posted, so please have a look on that as well.
Thank you!

AusBruce commented:
Review:

The replacement files are created successfully.
The old component is unlinked and new component is successfully added.
HTML, CoffeeScript, and SCSS are migrated successfully and no error
The website remain the same as picture shows
Great. It is good to have upstream review!

satikaj commented:
Great work so far @b0ink 

One thing I noticed when testing with an image, if it's too big it was scrollable in the original, and in the new version it just restricts it to the viewport size. Could you look into this?

Also, if possible, can you use Tailwind classes for the styles instead of SCSS

b0ink committed:
fix: allow oversized images to be scrollable

b0ink committed:
refactor: replace sass with tailwind css

b0ink commented:
> One thing I noticed when testing with an image, if it's too big it was scrollable in the original, and in the new version it just restricts it to the viewport size. Could you look into this?

@satikaj  Closest I could get to make it scrollable like the original from within the component's styling was adding scroll overflow on just the modal itself [8ff67d2](https://github.com/thoth-tech/doubtfire-web/pull/218/commits/8ff67d2f8e483813bc3af0dc2429b58fe16f8f2b) but if we wanted the same behaviour as bootstrap's modals where you can also scroll on the opaque backdrop we would have to add overflow to [.cdk-global-overlay-wrapper globally](https://stackblitz.com/edit/angular-ss6wkf-zzzqdt?file=index.html,main.ts,app%2Fdialog-overview-example-dialog.html,app%2Fdialog-overview-example.html,styles.css) - but this would break the click to close modal functionality

Also worth noting that the modal's box-shadow is lost due to this commit

satikaj commented:
I see. Should be fine since this is the closest we can get it.

Please make an upstream PR to doubtfire-lms/doubtfire-web 9.x branch @b0ink

b0ink committed:
refactor: only reveal scrollbar on oversized images

b0ink commented:
Made one small change to use `overflow-y-auto` instead of `overflow-y-scroll` so that the blank scrollbar doesn't show up unless the image is oversized and becomes scrollable

PR has been opened: https://github.com/doubtfire-lms/doubtfire-web/pull/869

b0ink committed:
feat: persistent pdf zoom level

b0ink committed:
refactor: replace bootstrap html with material components

b0ink committed:
refactor: improve modal size

b0ink committed:
fix: remove duplicate alert

b0ink committed:
refactor: align caption with modal title

AusBruce commented:
<img width="1280" alt="Screenshot 2024-09-08 at 11 45 13 AM" src="https://github.com/user-attachments/assets/fa179890-fffc-4caa-8d43-353cfda766f9">
Tested, works fine on my local machine

AusBruce commented:
Tested, if the image is oversize then will have a scrollbar.

<img width="1280" alt="Screenshot 2024-09-08 at 12 12 35 PM" src="https://github.com/user-attachments/assets/0e5f22d6-e103-4a3a-bafd-03717cfc70f7">

<img width="1280" alt="Screenshot 2024-09-08 at 12 12 50 PM" src="https://github.com/user-attachments/assets/8191de75-4080-4870-9a1b-d4c1e840158f">

b0ink committed:
Merge remote-tracking branch 'doubtfirelms/8.0.x' into migrate/comments-modal

b0ink committed:
refactor: replace iframe with native pdf-viewer

b0ink committed:
Merge remote-tracking branch 'doubtfirelms/8.0.x' into migrate/confirmation-modal

b0ink committed:
feat: add startpage property to auto scroll to on pdf render

b0ink committed:
feat: init pdf viewer on page 2 for student submissions

b0ink committed:
Merge remote-tracking branch 'doubtfirelms/8.0.x' into feat/persistent-pdf-zoom

macite committed:
fix: correct ability to edit teaching periods

macite committed:
fix: ensure an old grade is present in portfolio assessment

Fixes an issue where this somehow becomes null or undefined.

macite committed:
fix: ensure overseer images can be edited

macite committed:
chore(release): 8.0.35

macite committed:
fix: overseer image admin in unit and task

Correct connections and concurrency bug in loading overseer image when unit loaded.

macite committed:
chore: add proxy for running in dev using compose

macite committed:
chore(release): 8.0.36

macite committed:
fix: ensure overseer download get resources zip

macite committed:
Merge branch '8.0.x' of https://github.com/macite/doubtfire-web into 8.0.x

macite committed:
chore(release): 8.0.37

Lolretrorat committed:
fix: correct a typo in /edit_profile

macite committed:
feat: add option for institution logo

Embedded top left in toolbar.

CaptainBoggle committed:
fix: correct string manipulation for task period formatting

macite committed:
fix: reply to position and wrapping

macite committed:
chore: remove unused old angular tools

b0ink committed:
fix: support windows build

b0ink committed:
fix: support hot reloading on windows

b0ink committed:
chore: update package-lock.json

b0ink committed:
fix: update packages

b0ink committed:
chore: save esbuild to devdependencies

b0ink committed:
fix: lock ng2-pdf-viewer to 10.2.2

macite committed:
Merge branch 'fix/windows-build' of https://github.com/b0ink/doubtfire-web into b0ink-fix/windows-build

macite committed:
Merge branch 'b0ink-fix/windows-build' into 8.0.x

macite committed:
Merge branch 'bugfix/fix_typo_in_edit_profile' of https://github.com/Lolretrorat/doubtfire-web into Lolretrorat-bugfix/fix_typo_in_edit_profile

macite committed:
Merge branch 'Lolretrorat-bugfix/fix_typo_in_edit_profile' into 8.0.x

macite committed:
Merge branch 'feat/persistent-pdf-zoom' of https://github.com/b0ink/doubtfire-web into b0ink-feat/persistent-pdf-zoom

macite committed:
Merge branch 'b0ink-feat/persistent-pdf-zoom' into 8.0.x

macite committed:
Merge branch 'feat/pdfviewer-custom-start-page-8.0.x' of https://github.com/b0ink/doubtfire-web into b0ink-feat/pdfviewer-custom-start-page-8.0.x

macite committed:
Merge branch 'b0ink-feat/pdfviewer-custom-start-page-8.0.x' into 8.0.x

macite committed:
Merge branch 'migrate/confirmation-modal' of https://github.com/b0ink/doubtfire-web into b0ink-migrate/confirmation-modal

macite committed:
Merge branch 'b0ink-migrate/confirmation-modal' into 8.0.x

macite committed:
Merge branch 'development' of https://github.com/CaptainBoggle/doubtfire-web into 8.0.x

macite committed:
chore: remove arm specific package

b0ink committed:
Merge remote-tracking branch 'upstream/8.0.x' into migrate/comments-modal
